core/object/object.cpp

```diff
@@ -1130,15 +1131,23 @@ Error Object::emit_signalp(const StringName &p_name, const Variant **p_args, int
 		const Variant **args = p_args;
 		int argc = p_argcount;
 
-		if (c.binds.size()) {
+		bool autobound = c.flags & CONNECT_SOURCE_AUTOBIND;
+
+		if (c.binds.size() || autobound) {
+			int source_count = autobound ? 1 : 0;
+
 			//handle binds
-			bind_mem.resize(p_argcount + c.binds.size());
+			bind_mem.resize(source_count + p_argcount + c.binds.size());
```

Comment:
I think for clarity the appended object should come at the end, not at the begining.

Reply:
If so, then use `Callable.bind` in code to bind some parameters, it won't go at the end, just before those bound parameters.

Of course, for the **Connection** dialog, `Callable.bind` will not be involved. It will be at the end.

Reply:
Ah yeah, callable binds will come at the end, this makes sense.

Reply:
To be completely honest, I think we should remove connection binds and only use callable binds but this is quite a lot of work.

That said, I think the instance should come _after_ the regular function arguments (even if it comes before the callable bind). This is because we only have the concept of appending arguments to functions. If we had to start discerning between prepending and appending it would be a mess.